jenkins/Makefile

```diff
@@ -1,13 +1,77 @@
+#
 # A simple Makefile for building and pushing the various pbench-devel-*
-# container images.
+# and pbench-rpmbuild container images.
+#
+# This makefile defines the following targets:
 #
-# WARNING: Do not publish images layered on top of UBI images.
+#  all (the default):  builds the CI image and a Fedora-based development image
+#  image-ci:           builds the CI image
+#  image-fedora:       builds the Fedora-based development image
+#  image-rpmbuild-all: builds all the default containers for building RPMs
+#  image-rpmbuild-rhel-7:  builds a single RPM build container (e.g., for RHEL-7)
+#  push-ci:            pushes the CI image to the registry
+#  push-fedora:        pushes the Fedora-based development image to the registry
+#  push-rpmbuild-all:  pushes all the default RPM build containers to the registry
+#
+
+# Default list of distributions for which to build containers in which to build RPMs
+RPMBUILD_BASEIMAGE_DEFAULTS = \
+	rhel-9 rhel-8 rhel-7 \
+	fedora-36 fedora-35 fedora-34 \
+	centos-9 centos-8
+
+# Registry where the resulting RPM build container images are to be stored
+RPMBUILD_IMAGE_REPO = images.paas.redhat.com/pbench
+
+# Templates for the various distributions' base images.  For the moment, for a
+# given distribution, all of the versions can be found by tweaking the image
+# tag; so, to generate the appropriate repository for the base image, we just
+# have to pick the appropriate template and insert the requested version, which
+# is exactly what happens when RPMBUILD_BASEIMAGE_REPO is expanded.
 #
+# Note that registry.access.redhat.com is moving to registry.redhat.io where it
+# will no longer permit anonymous access, but it works for the moment.
+VER := __version__
+RPMBUILD_BASEIMAGE_rhel = registry.access.redhat.com/ubi${VER}:latest
+RPMBUILD_BASEIMAGE_fedora = quay.io/fedora/fedora:${VER}
+RPMBUILD_BASEIMAGE_centos = quay.io/centos/centos:stream${VER}
+RPMBUILD_BASEIMAGE_REPO = $(subst ${VER},${DIST_VERSION},${RPMBUILD_BASEIMAGE_${DIST_NAME}})
+
+# List of packages required to build RPMs, to be installed in the build
+# containers.  (The jinja2 CLI is not available as an RPM on RHEL, so we
+# install it via pip for all platforms, to keep things simple.)
```

Comment:
OK, I really hate stacked simultaneous PRs; but, fine, the comment belongs here. So just drop the whole "jinja2 cli" thing from this comment, because it's not nothing at all to do with your `RPMBUILD_PACKAGES` assignment. If you want to mention in the git log that you dropped it for this reason, fine, but it doesn't need to be permanently recorded in the code.

Unless, that is, you want to break this into a `TODO` to switch back to an RPM when and if it's available everywhere.

Maybe we need a floating portmanteau epic with stories "to review every release or so and schedule if certain trigger conditions have changed"...

Reply:
I hate stacked simultaneous PRs, too, but when we produce work faster than we review it, the PRs tend to stack up, one way or another.

The comment is intended to explain why `RPMBUILD_PACKAGES` does not contain the Jinja2 CLI package.  I suppose I could omit the part about Pip, but I'm not sure that that would address your concern.  Making it into a TODO would be reasonable (as would having a periodic task to reevaluate all of our dependencies...); I'll do tht if it doesn't end up creating an extra round of reviews.